frontend/console/src/features/traces/TraceDetailItem.tsx

```diff
@@ -0,0 +1,74 @@
+import type { Timestamp } from '@bufbuild/protobuf'
+import type { TraceEvent } from '../../api/timeline/use-request-trace-events'
+import { CallEvent, type Event, IngressEvent } from '../../protos/xyz/block/ftl/v1/console/console_pb'
+import { TimelineIcon } from '../timeline/TimelineIcon'
+import { eventBackgroundColor } from '../timeline/timeline.utils'
+import { eventBarLeftOffsetPercentage } from './traces.utils'
+
+interface TraceDetailItemProps {
+  event: Event
+  traceEvent: TraceEvent
+  eventDurationMs: number
+  requestDurationMs: number
+  requestStartTime: Timestamp | undefined
+  selectedEventId: bigint | undefined
+  handleEventClick: (eventId: bigint) => void
+}
+
+export const TraceDetailItem: React.FC<TraceDetailItemProps> = ({
+  event,
+  traceEvent,
+  eventDurationMs,
+  requestDurationMs,
+  requestStartTime,
+  selectedEventId,
+  handleEventClick,
+}) => {
+  const leftOffsetPercentage = eventBarLeftOffsetPercentage(event, requestStartTime, requestDurationMs)
+
+  let width = (eventDurationMs / requestDurationMs) * 100
+  if (width < 1) width = 1
+
+  let action = ''
+  let eventName = ''
+  const icon = <TimelineIcon event={event} />
+
+  if (traceEvent instanceof CallEvent) {
+    action = 'Call'
+    eventName = `${traceEvent.destinationVerbRef?.module}.${traceEvent.destinationVerbRef?.name}`
+  } else if (traceEvent instanceof IngressEvent) {
+    action = `HTTP ${traceEvent.method}`
+    eventName = `${traceEvent.path}`
+  }
+
+  let barColor = eventBackgroundColor(event)
+  if (event.id === selectedEventId) {
+    barColor = 'bg-green-500'
+  }
+
+  const isSelected = event.id === selectedEventId
+  const listItemClass = isSelected
+    ? 'flex items-center justify-between p-2 bg-indigo-100/50 dark:bg-indigo-700 rounded cursor-pointer'
+    : 'flex items-center justify-between p-2 hover:bg-indigo-500/10 rounded cursor-pointer'
```

Comment:
Since there's so much overlap between these two branch cases, what do you think about something like:

```
const listItemClass = classNames(isSelected ? '...' : '...', 'shared class names')
```

As is, it took me a minute to parse out the exact differences between the two, since there are some shared classes at both the head and tail ends